sockeye/utils.py

```diff
@@ -233,9 +233,47 @@ def smart_open(filename: str, mode: str = "rt", ftype: str = "auto", errors: str
     if ftype in ('gzip', 'gz') \
             or (ftype == 'auto' and filename.endswith(".gz")) \
             or (ftype == 'auto' and 'r' in mode and is_gzip_file(filename)):
-        return gzip.open(filename, mode=mode, encoding='utf-8', errors=errors)
+            if mode == "rb" or mode == "wb":
+                return gzip.open(filename, mode=mode)
+            else:
+                return gzip.open(filename, mode=mode, encoding='utf-8', errors=errors)
     else:
-        return open(filename, mode=mode, encoding='utf-8', errors=errors)
+        if mode == "rb" or mode == "wb":
+            return open(filename, mode=mode)
+        else:
+            return open(filename, mode=mode, encoding='utf-8', errors=errors)
+
+
+def combine_means(means: List[float], num_sents: List[int]) -> float:
```

Comment:
Thanks for extracting this logic into separate functions!

Given that floats could be `None` should the type be `List[Optional[float]]`?

Could you add a unit test for combine_means and combine_stds?

Reply:
```suggestion
def combine_means(means: List[Optional[float]], num_sents: List[int]) -> float:
```